File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const n = prices.length
    var answer = new Array(n).fill(0);
    const stack = [];
    
    stack.push(0);
    
    for(var i = 1; i < n; i++){
        while(stack.length > 0 && prices[stack[stack.length - 1]] > prices[i]){
            answer[stack[stack.length - 1]] = i - stack.pop();
        }
        
        stack.push(i);
    }
    
    while(stack.length){
        const a = stack.pop();
        answer[a] = n - a - 1;
    }
    
    return answer;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    
    l = len(prices)
    for i in range(l):
        for j in range(i+1, l):
            if prices[i] > prices[j] or j == l-1:
                answer.append(j-i)
                break
    answer.append(0)
    return answer